admin/import/plugins/class-import-aioseo.php

```diff
@@ -1,115 +1,97 @@
 /**
- * WPSEO plugin file.
+ * File with the class to handle data from All in One SEO Pack.
  *
  * @package WPSEO\Admin\Import\Plugins
  */
 
 /**
- * Class with functionality to import Yoast SEO settings from All In One SEO.
+ * Class with functionality to import & clean All in One SEO Pack post metadata.
  */
-class WPSEO_Import_AIOSEO implements WPSEO_Plugin_Importer {
+class WPSEO_Import_AIOSEO extends WPSEO_Plugin_Importer {
 	/**
-	 * @var wpdb Holds the WPDB instance.
+	 * @var string The plugin name
 	 */
-	protected $wpdb;
+	protected $plugin_name = 'All In One SEO Pack';
 
 	/**
-	 * Holds the import status object.
-	 *
-	 * @var WPSEO_Import_Status
-	 */
-	private $status;
-
-	/**
-	 * WPSEO_Import_AIOSEO constructor.
+	 * @var string Meta key, used in like clause for detect query.
 	 */
-	public function __construct() {
-		global $wpdb;
-
-		$this->wpdb = $wpdb;
-	}
+	protected $meta_key = '_aioseop_%';
 
 	/**
-	 * Returns the plugin name.
-	 *
-	 * @return string Plugin name.
+	 * @var array Array of keys to clone
 	 */
-	public function plugin_name() {
-		return 'All In One SEO Pack';
-	}
-
-	/**
-	 * Returns whether there is post meta data to import.
-	 *
-	 * @return WPSEO_Import_Status Import status object.
-	 */
-	public function detect() {
-		$this->status = new WPSEO_Import_Status( 'detect', false );
-
-		if ( ! $this->detect_helper() ) {
-			return $this->status;
-		}
-
-		return $this->status->set_status( true );
-	}
+	protected $clone_keys = array(
+		array(
+			'old_key' => '_aioseop_title',
+			'new_key' => 'title',
+		),
+		array(
+			'old_key' => '_aioseop_description',
+			'new_key' => 'metadesc',
+		),
+		array(
+			'old_key' => '_aioseop_noindex',
+			'new_key' => 'meta-robots-noindex',
+			'convert' => array( 'on' => 1 ),
+		),
+		array(
+			'old_key' => '_aioseop_nofollow',
+			'new_key' => 'meta-robots-nofollow',
+			'convert' => array( 'on' => 1 ),
+		),
+	);
 
 	/**
-	 * Imports the All in one SEO Pack settings.
+	 * Import All In One SEO meta values.
 	 *
-	 * @return WPSEO_Import_Status Import status object.
+	 * @return bool Import success status.
 	 */
-	public function import() {
-		$this->status = new WPSEO_Import_Status( 'import', false );
-
-		if ( ! $this->detect_helper() ) {
-			return $this->status;
+	protected function import() {
+		$status = parent::import();
+		if ( $status ) {
+			$this->import_opengraph();
+			return true;
 		}
-
-		$this->import_metas();
-
-		return $this->status->set_status( true );
+		return false;
 	}
 
 	/**
-	 * Removes the All in one SEO pack data from the database.
+	 * Imports the OpenGraph and Twitter settings for all posts.
 	 *
-	 * @return WPSEO_Import_Status Import status object.
+	 * @return bool
 	 */
-	public function cleanup() {
-		$this->status = new WPSEO_Import_Status( 'cleanup', false );
+	protected function import_opengraph() {
+		$query_posts = new WP_Query( 'post_type=any&meta_key=_aioseop_opengraph_settings&order=ASC&fields=ids&nopaging=true' );
```

Comment:
Is `nopaging=true` the best way to go here ? This can lead to a *huge* number of results.
Could a loop with 1.000 or 10.000 results each time help prevent potential "out of memory" issues and give a more stable experience ?

Note: this is used in several places across the import scripts, not all of which I can comment on based on the current diff.

Reply:
Yeah we probably should do this differently, but not in this pull :D